### PR TITLE
Fix doc comment references to 'this'

### PR DESCRIPTION
**Pull request by srawlins**
Work towards https://github.com/dart-lang/dartdoc/issues/3761

Sibling CL to https://dart-review.googlesource.com/c/sdk/+/365204

The analyzer has never recognized `[this]` as a valid doc comment
reference (and the `comment_references` lint rule has similarly
reported such reference attempts). dartdoc has its own algorithms
for resolving comment references, which we are dismantling in favor
of a single resolution, provided by the analyzer.

We've also decided against adding support in the analyzer (see
https://github.com/dart-lang/sdk/issues/58155), so these
reference attempts should be re-written.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 1]
```diff
@@ -3878,8 +3878,8 @@ class SemanticsConfiguration {
   /// This is a request to increase the value represented by the widget. For
   /// example, this action might be recognized by a slider control.
   ///
-  /// If [this.value] is set, [increasedValue] must also be provided and
-  /// [onIncrease] must ensure that [this.value] will be set to
+  /// If `this.value` is set, [increasedValue] must also be provided and
+  /// [onIncrease] must ensure that `this.value` will be set to
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart:3882]: I think this should be:

```suggestion
  /// If [value] is set, [increasedValue] must also be provided and
  /// [onIncrease] must ensure that [value] will be set to
```

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 2]
```diff
@@ -3897,8 +3897,8 @@ class SemanticsConfiguration {
   /// This is a request to decrease the value represented by the widget. For
   /// example, this action might be recognized by a slider control.
   ///
-  /// If [this.value] is set, [decreasedValue] must also be provided and
-  /// [onDecrease] must ensure that [this.value] will be set to
+  /// If `this.value` is set, [decreasedValue] must also be provided and
+  /// [onDecrease] must ensure that `this.value` will be set to
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart:3901]:
```suggestion
  /// If [value] is set, [decreasedValue] must also be provided and
  /// [onDecrease] must ensure that [value] will be set to
```

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 3]
```diff
@@ -4812,11 +4812,11 @@ class SemanticsConfiguration {
     _setFlag(SemanticsFlag.isReadOnly, value);
   }
 
-  /// Whether [this.value] should be obscured.
+  /// Whether `this.value` should be obscured.
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart:4815]:
```suggestion
  /// Whether [value] should be obscured.
```

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 4]
```diff
@@ -4812,11 +4812,11 @@ class SemanticsConfiguration {
...
   ///
   /// This option is usually set in combination with [isTextField] to indicate
   /// that the text field contains a password (or other sensitive information).
-  /// Doing so instructs screen readers to not read out [this.value].
+  /// Doing so instructs screen readers to not read out `this.value`.
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart:4819]:
```suggestion
  /// Doing so instructs screen readers to not read out [value].
```

File: packages/flutter/lib/src/semantics/semantics.dart [hunk 5]
```diff
@@ -4844,7 +4844,7 @@ class SemanticsConfiguration {
   }
 
   /// The currently selected text (or the position of the cursor) within
-  /// [this.value] if this node represents a text field.
+  /// `this.value` if this node represents a text field.
```
Comment (goderbauer) [inline packages/flutter/lib/src/semantics/semantics.dart:4847]:
```suggestion
  /// [value] if this node represents a text field.
```

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 1 above)
Comment (srawlins) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 2 above)
Comment (srawlins) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 3 above)
Comment (srawlins) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 4 above)
Comment (srawlins) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

File: packages/flutter/lib/src/semantics/semantics.dart (hunk 5 above)
Comment (srawlins) [reply packages/flutter/lib/src/semantics/semantics.dart]: Done

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**